File: data/derived_cot/rq3_prompting/results/translation/qwen/cpp_to_py/code_output/BitStatusUtil.py
class BitStatusUtil:
    @staticmethod
    def add(states, stat):
        BitStatusUtil.check_two_args(states, stat)
        return states | stat

    @staticmethod
    def has(states, stat):
        BitStatusUtil.check_two_args(states, stat)
        return (states & stat) == stat

    @staticmethod
    def remove(states, stat):
        BitStatusUtil.check_two_args(states, stat)
        if BitStatusUtil.has(states, stat):
            return states ^ stat
        return states

    @staticmethod
    def check_two_args(a, b):
        for num in (a, b):
            if num < 0:
                raise ValueError(f"{num} must be greater than or equal to 0")
            if num % 2 != 0:
                raise ValueError(f"{num} not even")

File: data/derived_cot/rq1_traces/translation/glm5.1/java_to_cpp/code_output/BitStatusUtil.cpp
#include <iostream>
#include <stdexcept>
#include <string>
#include <initializer_list>

class BitStatusUtil {
public:
    static int add(int states, int stat) {
        check({states, stat});
        return states | stat;
    }

    static bool has(int states, int stat) {
        check({states, stat});
        return (states & stat) == stat;
    }

    static int remove(int states, int stat) {
        check({states, stat});
        if (has(states, stat)) {
            return states ^ stat;
        }
        return states;
    }

    static void check(std::initializer_list<int> args) {
        for (int arg : args) {
            if (arg < 0) {
                throw std::invalid_argument(std::to_string(arg) + " must be greater than or equal to 0");
            }
            if (arg % 2 != 0) {
                throw std::invalid_argument(std::to_string(arg) + " not even");
            }
        }
    }
};

int main() {
    std::cout << BitStatusUtil::add(2, 4) << std::endl;
    std::cout << std::boolalpha << BitStatusUtil::has(6, 2) << std::endl;
    std::cout << BitStatusUtil::remove(6, 2) << std::endl;
    try {
        BitStatusUtil::check({2, 3, 4});
    } catch (const std::invalid_argument& e) {
        std::cout << e.what() << std::endl;
    }
    return 0;
}